### PR TITLE
Removes ES6 syntax to reactivate ESLint

### PR DESCRIPTION
**Pull request by tonytlwu**
The ES6 syntax is preventing ESLint from working as it's catching a series of syntax errors that aren't supported by ES5.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -359,7 +360,11 @@ function fetchCurrentDataSourceEntries(entries) {
     } else {
       var flattenedColumns = {};
 
-      rows.map(({ data }) => data).forEach(dataItem => (flattenedColumns = { ...flattenedColumns, ...dataItem }));
+      rows.map(function(row) {
```
Comment (tonytlwu) [inline js/interface.js:363]: @squallstar Can you confirm if this is still correctly optimised based on the work done in https://github.com/Fliplet/fliplet-widget-data-source/pull/177

**Review by squallstar — APPROVED** [review]

File: js/interface.js (hunk 1 above)
Comment (squallstar) [reply js/interface.js:363]: @tonytlwu is there a specific reason you want to change the current functionality not to use the spread operator?

Your solution seems faster but creates almost twice as many operations?

Comment (tonytlwu) [reply js/interface.js:363]: @squallstar Because it isn't allowed for ES5 and our ESLint has not been updated to support this.

Comment (tonytlwu) [reply js/interface.js:363]: I think high ops/s is meant to be a good thing though.

**Review by squallstar — APPROVED** [review]

**Pull request merged**